File: 05_ui_screens/main-screens/test_api.js
async function test() {
  const initRes = await fetch('https://aidiet-api.onrender.com/api/v1/auth/init', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ initData: 'dummy', platform: 'web' })
  });
  const authData = await initRes.json();
  const token = authData.token || authData.access_token;

  const res = await fetch('https://aidiet-api.onrender.com/api/v1/plan/generate', {
    method: 'POST',
    headers: { 
      'Content-Type': 'application/json',
      'Authorization': 'Bearer ' + token
    },
    body: JSON.stringify({
      "age": 30, "gender": "male", "weight": 80, "height": 180, "goal": "Сбалансированное питание",
      "activity_level": "Умеренная", "activity_multiplier": null, "allergies": [], "restrictions": [],
      "diseases": [], "symptoms": [], "medications": "Нет", "supplements": "Нет", "supplement_openness": null,
      "country": "Россия", "city": "", "budget_level": "Средний", "cooking_time": "Без разницы",
      "fasting_status": false, "meal_pattern": "3 приема (завтрак, обед, ужин)", "training_schedule": "Без регулярных тренировок",
      "sleep_schedule": "8 часов", "liked_foods": [], "disliked_foods": [], "excluded_meal_types": [],
      "motivation_barriers": [], "womens_health": null, "bmi": null, "waist": null, "body_type": null,
      "blood_tests": null, "tier": "T1"
    })
  });
  const data = await res.json();
  console.log("Data:", data);
}
test();

File: ejeweeka_backend/test_api.py
import asyncio
import httpx
import json

async def run():
    url = "http://127.0.0.1:8001/api/v1/plan/generate"
    headers = {"Content-Type": "application/json"}
    data = {
      "age": 46,
      "gender": "female",
      "weight": 65.0,
      "height": 168.0,
      "goal": "Адаптировать питание к возрасту",
      "activity_level": "2 раза в неделю",
      "target_weight": 56.0,
      "target_timeline_weeks": 13,
      "bmi": "23.0",
      "bmi_class": "normal",
      "waist": "76.0",
      "country": "Беларусь",
      "budget_level": "Экономный",
      "cooking_time": "Без разницы",
      "allergies": [],
      "restrictions": ["vegetarian", "no_red_meat", "pescatarian", "no_dairy"],
      "diets": ["vegetarian", "no_red_meat", "pescatarian", "no_dairy"],
      "diseases": ["insulin_resistance"],
      "symptoms": ["constipation", "unstable_stool", "sugar_cravings"],
      "womens_health": ["irregular_cycle", "breastfeeding"],
      "medications": "клайра",
      "supplements": "Нет",
      "fasting_type": "none",
      "tier": "gold"
    }

    async with httpx.AsyncClient(timeout=120) as client:
        response = await client.post(url, headers=headers, json=data)
        if response.status_code == 200:
            print(json.dumps(response.json(), indent=2, ensure_ascii=False))
        else:
            print(response.status_code, response.text)

asyncio.run(run())
